### PR TITLE
after printing to open the cash box

### PR DESCRIPTION
**Pull request by dracenliu**

### CODE REVIEW

File: escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinter.java [hunk 1]
```diff
@@ -156,6 +156,7 @@ public EscPosPrinter printFormattedTextAndCut(String text, int dotsFeedPaper) th
 
         this.printFormattedText(text, dotsFeedPaper);
         this.printer.cutPaper();
+        this.printer.openCashBox();
```
Comment (DantSu) [inline escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinter.java:159]: Remove this.
Add a new method `printFormattedTextAndOpenCashBox(String text, int dotsFeedPaper)` like this : 

```java
    /**
     * Print a formatted text, cut the paper and open the cash box. Read the README.md for more information about text formatting options.
     *
     * @param text Formatted text to be printed.
     * @param dotsFeedPaper distance feed paper at the end.
     * @return Fluent interface
     */
    public EscPosPrinter printFormattedTextAndOpenCashBox(String text, int dotsFeedPaper) throws EscPosConnectionException, EscPosParserException, EscPosEncodingException, EscPosBarcodeException {
        if (this.printer == null || this.printerNbrCharactersPerLine == 0) {
            return this;
        }

        this.printFormattedTextAndCut(text, dotsFeedPaper);
        this.printer.openCashBox();
        return this;
    }
```

**Review by DantSu — CHANGES_REQUESTED** [review]

File: escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinterCommands.java [hunk 2]
```diff
@@ -446,6 +446,22 @@ public EscPosPrinterCommands feedPaper(int dots) throws EscPosConnectionExceptio
...
+     *
+     * @return Fluent interface
+     * @throws EscPosConnectionException
+     */
+    public EscPosPrinterCommands openCashBox() throws EscPosConnectionException {
+        if (!this.printerConnection.isConnected()) {
+            return this;
+        }
+
+        this.printerConnection.write(new byte[]{0x1B, 0x70, 0x00, 0x3C, (byte) 0xFF});
```
Comment (DantSu) [inline escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinterCommands.java:461]: You can remove `(byte)`

**Review by DantSu — CHANGES_REQUESTED** [review]

File: escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinterCommands.java (hunk 2 above)
Comment (dracenliu) [reply escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinterCommands.java:461]: 0xff out of byte value,so must cast to byte

File: escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinter.java (hunk 1 above)
Comment (dracenliu) [reply escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinter.java:159]: ok,good idea!

File: escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinterCommands.java (hunk 2 above)
Comment (DantSu) [reply escposprinter/src/main/java/com/dantsu/escposprinter/EscPosPrinterCommands.java:461]: Sorry, yes you right, forget it ^^

**Pull request closed without merging**